[Training Manual] 1/3 Chapter 7 (#3109)
qgis/QGIS-Documentation: source/docs/training_manual/vector_analysis/reproject_transform.rst

|basic| |FA| "On the Fly" Reprojection
-------------------------------------------------------------------------------

QGIS **always** reprojects data "on the fly". What this means is that even if
the data itself is in another CRS, QGIS can project it as if it were in a CRS
of your choice.

> You can change the projection CRS of the project by clicking on |projectionEnabled|

Reviewer: remove "projection" in front of CRS.

button in the bottom right corner of QGIS.

* In the dialog that appears, type the word ``global`` into the :guilabel:`Filter`
  field. One CRS (:guilabel:`NSIDC EASE-Grid Global`) should appear in the list
  below.
* Click on the :guilabel:`NSIDC EASE-Grid Global` to select it, then click
  **[[OK]]**.
* Notice how the shape of South Africa changes. All projections work by
  changing the apparent shapes of objects on Earth.
* Zoom in to a scale of :guilabel:`1:5000000` again, as before.
* Pan around the map.
* Notice how the scale stays the same!

"On the fly" reprojection is also used for combining datasets that are in
different CRSs.

* Add another vector layer to your map which has the data for South Africa
  only. You'll find it as :file:`exercise_data/world/RSA.shp`.

What do you notice?

> The layer is visible even if it has a different CRS then the :guilabel:`continents`

Reviewer: different CRS from the (then -> from)

one.

|moderate| |FA| Saving a Dataset to Another CRS
-----------------------------------------------

Sometimes you need to export an existing dataset in another CRS. As we will see

> in the next lesson, if you need to make some distance calculation on layer, it

Reviewer: some distance calculations -> distance calculations

> is always better to have the layer in a PCS.

Reviewer: Is PCS a well know abbreviation?

Be aware that the 'on the fly' reprojection is related to the **project** and not
to single layers. You can have many layers with many different CRS loaded in QGIS.
Therefore you need to set a new CRS for each layer you are interested to and you

> can do this only by **saving** the layer as a new layer.

Reviewer: I think this paragraph (from "Be aware" to "new layer.") should be removed.  It is not so relevant, and I think it will cause confusion.

Author: I think not. From the moment that we have always OTF projection now, I think that users have to know this kind of information.

Reviewer: Then it needs to be reformulated.  I don't think it is useful to mention 'on the fly' reprojection.
And what does "for each layer you are interested to" mean?

* Right-click on the :guilabel:`buildings` layer in the :guilabel:`Layers` panel.
* Select :guilabel:`Export --> Save Features As...` in the menu that appears. You
   will be shown the :guilabel:`Save vector layer as...` dialog.
* Click on the :guilabel:`Browse` button next to the :guilabel:`File name` field.
* Navigate to :file:`exercise_data/` and specify the name of the new layer as
  :guilabel:`buildings_reprojected.shp`.

> * We must change the value of the :guilabel:`CRS`. Only the recent CRS used will

Reviewer: Only recently used CRSs will

  be shown in the drop down menu.

> * Click on the |setProjection| button next dropdown menu.

Reviewer: button next to the dropdown  (add "to the ")

QGIS **always** reprojects data "on the fly". What this means is that even if
the data itself is in another CRS, QGIS can project it as if it were in a CRS
of your choice.

You can change the CRS of the project by clicking on |projectionEnabled| button
in the bottom right corner of QGIS.

* In the dialog that appears, type the word ``global`` into the :guilabel:`Filter`
  field. One CRS (:guilabel:`NSIDC EASE-Grid Global`) should appear in the list
  below;
* Click on the :guilabel:`NSIDC EASE-Grid Global` to select it, then click
  :guilabel:`OK`;
* Notice how the shape of South Africa changes. All projections work by
  changing the apparent shapes of objects on Earth.
* Zoom in to a scale of :guilabel:`1:5000000` again, as before;
* Pan around the map;
* Notice how the scale stays the same!

"On the fly" reprojection is also used for combining datasets that are in
different CRSs.

* Add another vector layer to your map which has the data for South Africa
  only. You'll find it as :file:`exercise_data/world/RSA.shp`.

What do you notice?

The layer is visible even if it has a different CRS from the :guilabel:`continents`
one.

> 

Author: I still don't fully agree. In QGIS 2 if you replicate this exercise than the layer RSA is not visible until you enable the OTF function. Now once you load RSA layer it is immediately visible and correctly placed.

|basic| |FA| "On the Fly" Reprojection
-------------------------------------------------------------------------------

By default, QGIS reprojects reprojects data "on the fly". What this means is that
even if the data itself is in another CRS, QGIS can project it as if it were in
a CRS of your choice.

You can change the CRS of the project by clicking on |projectionEnabled| button
in the bottom right corner of QGIS.

* In the dialog that appears, type the word ``global`` into the :guilabel:`Filter`
  field. One CRS (:guilabel:`NSIDC EASE-Grid 2.0 Global`, EPSG:6933) should
  appear in the list below;

> * Click on the :guilabel:`NSIDC EASE-Grid Global` to select it, then click

Reviewer 2: ```suggestion
* Click on the :guilabel:`NSIDC EASE-Grid 2.0 Global` to select it, then click
```

  :guilabel:`OK`;
* Notice how the shape of South Africa changes. All projections work by
  changing the apparent shapes of objects on Earth;
* Zoom in to a scale of :guilabel:`1:5000000` again, as before;
* Pan around the map;
* Notice how the scale stays the same!

"On the fly" reprojection is also used for combining datasets that are in
different CRSs.

* Add another vector layer to your map which has the data for South Africa
  only. You'll find it as :file:`exercise_data/world/RSA.shp`.
* Load it and a quick way to see what is its CRS is by hovering the mouse over
  the layer in the legend. It is ``EPSG:3410``.

What do you notice?

The layer is visible even if it has a different CRS from the :guilabel:`continents`
one.

|moderate| |FA| Saving a Dataset to Another CRS
-----------------------------------------------

Sometimes you need to export an existing dataset in another CRS. As we will see
in the next lesson, if you need to make some distance calculations on layer, it
is always better to have the layer in a projected coordinate system.

> Be aware that the 'on the fly' reprojection is related to the **project** and not

Reviewer: ```suggestion
Due to 'on the fly' reprojection,
```

> to single layers. This means that layers can have different CRS from the project

Reviewer: ```suggestion
layers does not have to have the same CRS as the project
```

even if you see them in the *correct* position.

> But you can easily export the layer in another CRS.

Reviewer: ```suggestion
You can export a layer in any supported CRS.
```
